litellm/caching.py

```diff
@@ -998,111 +997,70 @@ def _get_cache_logic(self, cached_response: Any):
-        ]
-
-        # Add more data
-        keys = self.index.load(new_data)
+        # store in redis semantic cache
+        self.llmcache.store(
+            prompt=prompt,
+            response=value
+        )
 
         return
 
     def get_cache(self, key, **kwargs):
         print_verbose(f"sync redis semantic-cache get_cache, kwargs: {kwargs}")
-        import numpy as np
-        from redisvl.query import VectorQuery
 
-        # query
         # get the messages
         messages = kwargs["messages"]
         prompt = "".join(message["content"] for message in messages)
 
-        # convert to embedding
-        embedding_response = litellm.embedding(
-            model=self.embedding_model,
-            input=prompt,
-            cache={"no-store": True, "no-cache": True},
-        )
-
-        # get the embedding
-        embedding = embedding_response["data"][0]["embedding"]
-
-        query = VectorQuery(
-            vector=embedding,
-            vector_field_name="litellm_embedding",
-            return_fields=["response", "prompt", "vector_distance"],
-            num_results=1,
-        )
+        # check the cache
+        results = self.llmcache.check(prompt=prompt)
 
-        results = self.index.query(query)
-        if results == None:
+        # handle results / cache hit
+        if not results:
             return None
-        if isinstance(results, list):
-            if len(results) == 0:
-                return None
 
-        vector_distance = results[0]["vector_distance"]
-        vector_distance = float(vector_distance)
+        cache_hit = results[0]
+        vector_distance = float(cache_hit["vector_distance"])
         similarity = 1 - vector_distance
-        cached_prompt = results[0]["prompt"]
+        cached_prompt = cache_hit["prompt"]
+        cached_response = cache_hit["response"]
 
-        # check similarity, if more than self.similarity_threshold, return results
         print_verbose(
-            f"semantic cache: similarity threshold: {self.similarity_threshold}, similarity: {similarity}, prompt: {prompt}, closest_cached_prompt: {cached_prompt}"
+            f"got a cache hit: similarity threshold: {self.similarity_threshold}, similarity: {similarity}, current_prompt: {prompt}, closest_cached_prompt: {cached_prompt}"
         )
-        if similarity > self.similarity_threshold:
-            # cache hit !
-            cached_value = results[0]["response"]
-            print_verbose(
-                f"got a cache hit, similarity: {similarity}, Current prompt: {prompt}, cached_prompt: {cached_prompt}"
-            )
-            return self._get_cache_logic(cached_response=cached_value)
-        else:
-            # cache miss !
-            return None
+        return self._get_cache_logic(cached_response=cached_response)
 
-        pass
 
     async def async_set_cache(self, key, value, **kwargs):
-        import numpy as np
+        # TODO - patch async support in redisvl for SemanticCache
```

Comment:
Should this todo be merged/public?

Reply:
so redis semantic cache is not async on this PR @tylerhutcherson ?

Reply:
The `SemanticCache` class in redisvl library is not async-supported yet. This will be on our roadmap for release soon. However, this `async_set_cache` method still works and tests pass as expected. 

Given all of the issues associated with the `redisvl==0.0.7` pin you all have now (non stable version), it would be good to get this out and solve all of those open issues. We can prioritize adding in async to our semantic cache class asap. Thoughts @ishaan-jaff ?